linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1113,7 +1121,14 @@ public boolean moveNext() {
               final TSource outer = outers.current();
               final Enumerable<TInner> innerEnumerable;
               if (outer == null) {
-                innerEnumerable = null;
```

Comment:
Since the `innerLookup` has only one entry (maybe `EmptyList` or some constant numbers in `Linq4jTest.ONE_SELECTOR`, etc.), can we just remove the branch (`outer==null`) to avoid introducing the variable `isConditionAlwaysTrue`? The logic in `else` branch can handle this case as the `outerKey` is also `EmptyList`.

Reply:
We can remove the branch `outer==null` only if 
1. the `outerKeySelector.apply(outer)` can handle null key, 
2. and we also should match the the `outerKey` type to know if this is a always true match,  
These 2 conditions are too dependent on the implementation and can not be guaranteed.

Reply:
I see, thanks!